src/lib.rs

```diff
@@ -111,7 +91,7 @@ impl TransClient {
     ///     Ok(())
     /// }
     /// ```
-    pub async fn session_get(&self) -> Result<RpcResponse<SessionGet>> {
+    pub async fn session_get(&mut self) -> Result<RpcResponse<SessionGet>> {
```

Comment:
Why should we allow to mutate the client within the instance? That looks dangerous